packages/mobile/src/navigator/QRNavigator.tsx

```diff
@@ -0,0 +1,136 @@
+import {
+  createMaterialTopTabNavigator,
+  MaterialTopTabBarProps,
+} from '@react-navigation/material-top-tabs'
+import { useIsFocused } from '@react-navigation/native'
+import React, { useEffect, useMemo, useRef, useState } from 'react'
+import { useAsync } from 'react-async-hook'
+import { useTranslation } from 'react-i18next'
+import { Dimensions, Platform, StatusBar, StyleSheet } from 'react-native'
+import { check, PERMISSIONS, RESULTS } from 'react-native-permissions'
+import Animated, { call, greaterThan, onChange } from 'react-native-reanimated'
+import { ScrollPager } from 'react-native-tab-view'
+import { Namespaces } from 'src/i18n'
+import { Screens } from 'src/navigator/Screens'
+import QRCode from 'src/qrcode/QRCode'
+import QRScanner from 'src/qrcode/QRScanner'
+import QRTabBar from 'src/qrcode/QRTabBar'
+import { SVG } from 'src/send/actions'
+import { ExtractProps } from 'src/utils/typescript'
+
+const Tab = createMaterialTopTabNavigator()
+
+const width = Dimensions.get('window').width
+const initialLayout = { width }
+
+type AnimatedScannerSceneProps = ExtractProps<typeof QRScanner> & {
+  position: Animated.Value<number>
+}
+
+// Component doing our custom transition for the QR scanner
+function AnimatedScannerScene({ position, ...props }: AnimatedScannerSceneProps) {
+  const isFocused = useIsFocused()
+  const [wasFocused, setWasFocused] = useState(isFocused)
+  const [isPartiallyVisible, setIsPartiallyVisible] = useState(false)
+  const cameraPermission = useAsync(check, [
+    Platform.select({ ios: PERMISSIONS.IOS.CAMERA, default: PERMISSIONS.ANDROID.CAMERA }),
```

Comment:
why default? and not android?

Reply:
yes I feel you, TypeScript complains otherwise because `Platform.select` return type would also contain `undefined`. Which is not accepted by the `check` function.

Reply:
ahh.